common/network-common/src/main/java/org/apache/spark/network/server/OneForOneStreamManager.java

```diff
@@ -130,7 +141,7 @@ public void connectionTerminated(Channel channel) {
 
         try {
           // Release all remaining buffers.
-          while (state.buffers.hasNext()) {
+          while (state.isBufferMaterializedOnNext && state.buffers.hasNext()) {
```

Comment:
Shouldn't it be `!state.isBufferMaterializedOnNext`?

Reply:
Done.